test/unit/data-types/enum.test.ts

```diff
@@ -0,0 +1,43 @@
+import { DataTypes } from '@sequelize/core';
+import { expect } from 'chai';
+import { sequelize } from '../../support';
+
+describe('DataTypes.ENUM', () => {
+  it('raises an error if no values are defined', () => {
+    expect(() => {
+      sequelize.define('omnomnom', {
+        bla: { type: DataTypes.ENUM },
+      });
+    }).to.throwWithCause(Error, 'DataTypes.ENUM cannot be used without specifying its possible enum values.');
+  });
+
+  describe('setting values for enums', () => {
+    it('should not throw if `values` is a readonly array', () => {
+      const Array = ['A', 'B', 'C', 'D', 'E'] as const;
+      const type = DataTypes.ENUM({ values: Array });
+
+      expect(type.options.values).to.be.equal(Array);
```

Comment:
these should be deep-equal but not identity-equal. We should clone the `values` array so users modifying the array doesn't cause unexpected bugs

Reply:
Unresolved because adding an expect is good but it must not be `.equal`, it must be `deep.equal`, the test should be this:

```ts
expect(type.options.values).to.deep.equal(values);
expect(type.options.values).not.to.equal(values);
```

Not quite the initial goal of this PR but normalizing the option bag will require cloning the array anyway so we might as well

Reply:
> 

Added and tested, now const values was not to equal mutable values

Reply:
Commit: https://github.com/sequelize/sequelize/pull/15256/commits/8c14e7a7bf890680348bb39f7d4a90daaa879fcd